kubespawner/spawner.py

```diff
@@ -444,8 +444,8 @@ def _deprecated_changed(self, change):
 
     uid = Union(
         trait_types=[
-            Integer(),
-            Callable(),
+            Integer(allow_none=True),
```

Comment:
Can you move the `allow_none` to the uid trait itself, rather than to the individual types in the union trait? I think that's the better way to do this.